dot/parachain/pvf/worker_pool.go

```diff
@@ -0,0 +1,212 @@
+package pvf
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+)
+
+const (
+	maxRequestsAllowed uint = 60
+)
+
+type validationWorkerPool struct {
+	mtx sync.RWMutex
+	wg  sync.WaitGroup
+
+	workers map[parachaintypes.ValidationCodeHash]*worker
```

Comment:
A validation code can be mapped to a parachain, rather than a candidate. Multiple candidates can have the same validation code.
This would mean that we will have only one worker per parachain, which is not ideal. We should be able to validate multiple candidate from the same parachain.

https://paritytech.github.io/polkadot-sdk/book/runtime-api/validation-code.html

Reply:
I'm using ValidationCodeHash as key for these workers since the worker is storing a runtime instance based on the validation code, we may update this key in the future for handling multiple instances of candidates.

Reply:
I am starting to wonder if two parachains can have the same validation code. I have a feeling that it is possible. In which case we will have to change this a bit. but let's evaluate and tackle that as a separate piece of work.